webv2/index/index.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+	utilities "github.com/cloudspannerecosystem/harbourbridge/webv2/utilities"
+)
+
+// IndexSuggestion adds redundant index issue and interleved index suggestion in issues and suggestions tab.
+func IndexSuggestion() {
+
+	sessionState := session.GetSessionState()
+
+	for _, spannerTable := range sessionState.Conv.SpSchema {
+		CheckIndexSuggestion(spannerTable.Indexes, spannerTable)
+	}
+}
+
+// Helper method for checking Index Suggestion.
+func CheckIndexSuggestion(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
+
+	checkRedundantIndex(index, spannerTable)
+	checkInterleaveIndex(index, spannerTable)
+}
+
+// redundantIndex check for redundant Index.
+// If present adds Redundant as an issue in Issues.
+func checkRedundantIndex(index []ddl.CreateIndex, spannerTable ddl.CreateTable) {
```

Comment:
Modified an index to be on column with PK from UI but the redundant msg didn't show up

Reply:
same issue while removing